### ISSUE TITLE
(Nuxt) Allow specifying container selector instead of container element

### ISSUE THREAD

**Issue by rchl** [Maronato/vue-toastification#70]
When specifying module options for `vue-toastification` in Nuxt, we can't provide HTML Element for `container` option at this point as DOM doesn't exist yet. The plugin could allow providing a CSS selector for the container element.

## Detailed Description
A possible and easily applicable workaround right now is to use `$toast.updateDefaults` to specify container element but that means that this has to be done in a different place and makes options more scattered.
Allowing to specify CSS selector would allow for having all default toast options in one place and have a clear overview of all that were applied.

## Possible Implementation
From types perspective it could look like this:
```
  /**
   * Container where the toasts are mounted.
   *
   * Can either be an element or CSS selector that will be evaluated on showing the toast.
   */
  container?: HTMLElement | string
```

**Comment by Maronato** [comment]
A container selector is a bad idea because it may not be mounted by the time the plugin initializes. If that happens, the resulting object from `document.querySelector` is `null` so the plugin will never be mounted on the page.

Not only that, but given that Vue apps may be used as web components, `document.querySelector` may not have access to the node provided and cause hard-to-debug errors.

What I can do is make `container` possibly be an async callback that returns a valid node. That way, the user is responsible for making sure the node exists before mounting the container.

Your callback may then use a `MutationObserver` and look like:
```js
function callback() {
  return new Promise(resolve => {
    const observer = new MutationObserver(function(mutations, me) {
      const myContainer = document.getElementById("toast-container");
      if (myContainer) {
        me.disconnect();
        resolve(myContainer);
      }
    });
    observer.observe(document, {
      childList: true,
      subtree: true
    });
  });
}
// ...
Vue.use(Toast, { container: callback });
```

This will only mount the plugin once an element with id `toast-container` appears on the page.

I've tested a possible implementation and I was able to keep the `this.$toast` API fully functional even with the plugin unmounted. The result is that every call to `this.$toast` is put on a queue that is executed when `toast-container` is mounted. This way, even though the container may not exist at first, all toasts will be rendered when it appears.
